cmd/deploy/start.go

```diff
@@ -41,6 +46,13 @@ type FormattableDeployStartResponse struct {
 	err             error
 }
 
+type ArmoryDeployClient interface {
+	PipelineStatus(ctx context.Context, pipelineID string) (*de.PipelineStatusResponse, *_nethttp.Response, error)
+	DeploymentStatus(ctx context.Context, deploymentID string) (*de.DeploymentStatusResponse, *_nethttp.Response, error)
+	StartPipeline(ctx context.Context, options deployment.StartPipelineOptions) (*de.StartPipelineResponse, *_nethttp.Response, error)
+	GetArmoryCloudClient() *armoryCloud.Client
+}
```

Comment:
The primary reason for the introduction of the interface and some refactoring around it was to enable the use of the introduced MockDeployClient. I feel the most important part of test validation for this feature is to check the request headers and body, which wasn't possible before this PR.